libelement/src/lmnt/compiler_state.hpp

```diff
@@ -61,6 +71,23 @@ struct stack_allocation : std::enable_shared_from_this<stack_allocation>
     {
         return rel_index + (parent ? parent->index() : 0);
     }
+
+    void set_executed_in(execution_type type)
+    {
+        executed_in = (std::max)(type, executed_in);
+    }
+};
+
+
+struct execution_context
+{
+    execution_context(execution_context* p, execution_type t) : parent(p), rel_type(t) {}
+
+    execution_context* const parent;
+    const execution_type rel_type;
+    std::unordered_set<const stack_allocation*> allocations;
+
+    execution_type type() const { return (parent ? (std::min)(rel_type, parent->type()) : rel_type); }
```

Comment:
the parenthesis around `std::min` are interesting, does this do something? bypass the windows min macro?

Reply:
Yep, exactly that!

Reply:
fancy 😁